contracts/governance/extensions/GovernorSequentialProposalId.sol

```diff
@@ -0,0 +1,46 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.20;
+
+import {Governor} from "../Governor.sol";
+
+abstract contract GovernorSequentialProposalId is Governor {
+    uint256 private _proposalCount;
+    mapping(uint256 proposalHash => uint256 proposalId) private _proposalIds;
+
+    function getProposalId(
+        address[] memory targets,
+        uint256[] memory values,
+        bytes[] memory calldatas,
+        bytes32 descriptionHash
+    ) public view virtual override returns (uint256) {
+        uint256 proposalHash = hashProposal(targets, values, calldatas, descriptionHash);
+
+        uint256 storedProposalId = _proposalIds[proposalHash];
+        return storedProposalId == 0 ? (_proposalCount + 1) : storedProposalId;
```

Comment:
IMO returning `_proposalCount+1` is dangerous here. You might call that function, get this value, assume its good and store it ... only to realize that when it ends up being proposed it received a different id.

I think we should rather return an error.

```suggestion
        uint256 storedProposalId = _proposalIds[proposalHash];
        if (storedProposalId == 0) revert GovernorNonexistentProposal(0);
        return storedProposalId;
```